lib/src/segy.c

```diff
@@ -801,6 +881,11 @@ static int set_field( char* header,
             memcpy( header + (fd->field_index - 1), &(fv.u8), formatsize( fd->datatype ));
             return SEGY_OK;
 
+        case SEGY_IEEE_FLOAT_8_BYTE:
+            fv.u64 = htobe64( fv.u64 );
```

Comment:
Not sure if it is related to `fd->value.u64 = 0;` discussion.

I wonder whether floating point numbers should have their own `htobe`/`betoh`. 

New version of endiannes library (the one we are using in dlisio and maybe would one day want to switch in segyio too???) does c-conversion with the help of union
https://gist.github.com/jtbr/7a43e6281e6cca353b33ee501421860c#file-endianness-h-L146

Though if you do not experience any issues with using uint htobe for float, then there is probably no need to do anything about it, at least yet.